niceml/data/featurecombiners/featurecombiner.py

```diff
@@ -0,0 +1,22 @@
+"""Module for feature combiner"""
+from abc import ABC, abstractmethod
+
+import pandas as pd
+
+
+class FeatureCombiner(ABC):
+    """FeatureCombiner that can combine column-based features as part
+    of a DfDataset to create new features."""
+
+    @abstractmethod
+    def combine_features(self, data: pd.DataFrame) -> pd.DataFrame:
+        """
+        The combine_features function takes in a dataframe and returns a new dataframe
```

Comment:
Where do the new features come from?

Reply:
I have added information about where the functions come from.